mergify_engine/tests/unit/test_badges.py

```diff
@@ -14,29 +14,36 @@
 # License for the specific language governing permissions and limitations
 # under the License.
 
+from starlette import testclient
+
 from mergify_engine import web
 
 
 def test_badge_redirect():
-    with web.app.test_request_context("/"):
-        reply = web._get_badge_url("mergifyio", "mergify-engine", "png")
-        assert reply.status_code == 302
+    with testclient.TestClient(web.app) as client:
+        reply = client.get(
+            "/badges/mergifyio/mergify-engine.png", allow_redirects=False
+        )
+        assert reply.status_code == 307
```

Comment:
It sounds to me that 302 was the right choice, no? Why changing, why here, why now? 🤔

Reply:
Right, the default between flask and starlette differ, I fixed it.